lib/Runtime/Language/InterpreterStackFrame.cpp

```diff
@@ -1746,6 +1746,46 @@ namespace Js
 #pragma optimize("", on)
 #endif
 
+    const bool InterpreterStackFrame::ShouldDoProfile(FunctionBody* executeFunction)
+    {
+        const bool doProfile = executeFunction->GetInterpreterExecutionMode(false) == ExecutionMode::ProfilingInterpreter ||
+            (executeFunction->IsInDebugMode() && DynamicProfileInfo::IsEnabled(executeFunction));
+        return doProfile;
+    }
+
+    InterpreterStackFrame* InterpreterStackFrame::CreateInterpreterStackFrameForGenerator(ScriptFunction* function, FunctionBody* executeFunction, JavascriptGenerator* generator, bool doProfile)
+    {
+        //
+        // Allocate a new InterpreterStackFrame instance on the recycler heap.
+        // It will live with the JavascriptGenerator object.
+        //
+        ScriptContext* functionScriptContext = function->GetScriptContext();
+        Arguments generatorArgs = generator->GetArguments();
+        InterpreterStackFrame::Setup setup(function, generatorArgs);
+        size_t varAllocCount = setup.GetAllocationVarCount();
+        size_t varSizeInBytes = varAllocCount * sizeof(Var);
+        DWORD_PTR stackAddr = reinterpret_cast<DWORD_PTR>(&generator); // as mentioned above, use any stack address from this frame to ensure correct debugging functionality
```

Comment:
I see that this code was outlined from further down, but the comment on this line seem to have lost context

Reply:
fixed up comment.